sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -207,6 +225,25 @@ case object PostgresDialect extends JdbcDialect {
       Some(StringType)
     } else if (sqlType == Types.OTHER && typeName.equals("jsonb")) {
       Some(StringType)
+    } else if (sqlType == Types.OTHER && typeName.equals("uuid")) {
+        Some(StringType)
+    } else if (sqlType == Types.ARRAY) {
+      typeName match {
```

Comment:
Does we need underscores in the head of `typeName`?
I quickly checked actual strings returned by `ResultSetMetaData#getColumnTypeName` in postgresql-jdbc, and
I found that they have no underscore.